
# Stanley Controller

# At initialisation
# :param control_gain:                (float) time constant [1/s]
# :param softening_gain:              (float) softening gain [m/s]
# :param yaw_rate_gain:               (float) yaw rate gain [rad]
# :param steering_damp_gain:          (float) steering damp gain
# :param max_steer:                   (float) vehicle's steering limits [rad]
# :param wheelbase:                   (float) vehicle's wheelbase [m]
# :param path_x:                      (numpy.ndarray) list of x-coordinates along the path
# :param path_y:                      (numpy.ndarray) list of y-coordinates along the path
# :param path_yaw:                    (numpy.ndarray) list of discrete yaw values along the path
# :param dt:                          (float) discrete time period [s]

# At every time step
# :param x:                           (float) vehicle's x-coordinate [m]
# :param y:                           (float) vehicle's y-coordinate [m]
# :param yaw:                         (float) vehicle's heading [rad]
# :param target_velocity:             (float) vehicle's velocity [m/s]
# :param steering_angle:              (float) vehicle's steering angle [rad]

# :return limited_steering_angle:     (float) steering angle after imposing steering limits [rad]
# :return target_index:               (int) closest path index
# :return crosstrack_error:           (float) distance from closest path index [m]


stanley_node:
  ros__parameters:
    control_gain: 2.5
    softening_gain: 1.0
    yaw_rate_gain: 0.0
    steering_damp_gain: 0.0
    max_steer: 0.41
    wheelbase: 0.30
    dt: 0.01
    x: [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 6.0,6.0,6.0,6.0,6.0,6.0,6.0, 5.0,4.0,3.0,2.0,1.0,0.0,-1.0,-2.0,-3.0,-4.0,-5.0,-6.0,-7.0,-7.0,-7.0,-7.0,-7.0,-7.0,-7.0,-7.0, -6.0,-5.0,-4.0,-3.0,-2.0]
    y: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0,2.0,3.0,4.0,5.0,6.0,7.0, 7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0, 6.0,5.0,4.0,3.0,2.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0]

waypoint_visualizer_node:
   ros__parameters:
    x: [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 6.0,6.0,6.0,6.0,6.0,6.0,6.0, 5.0,4.0,3.0,2.0,1.0,0.0,-1.0,-2.0,-3.0,-4.0,-5.0,-6.0,-7.0,-7.0,-7.0,-7.0,-7.0,-7.0,-7.0,-7.0, -6.0,-5.0,-4.0,-3.0,-2.0]
    y: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0,2.0,3.0,4.0,5.0,6.0,7.0, 7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0,7.0, 6.0,5.0,4.0,3.0,2.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0]